src/GraphQL/SchemaExtensions.cs

```diff
@@ -9,6 +11,21 @@ namespace GraphQL
     /// </summary>
     public static class SchemaExtensions
     {
+        /// <summary>
+        /// Enables some experimental features that are not in the official specification, i.e. ability to expose
+        /// user-defined meta-information via introspection. See https://github.com/graphql/graphql-spec/issues/300
+        /// for more information.
+        /// </summary>
+        /// <param name="schema">The schema for which the features are enabled.</param>
+        /// <param name="mode">Experimental features mode.</param>
+        /// <returns>Reference to the provided <paramref name="schema"/>Experimental features mode.</returns>
+        public static TSchema EnableExperimentalIntrospectionFeatures<TSchema>(this TSchema schema, ExperimentalIntrospectionFeaturesMode mode = ExperimentalIntrospectionFeaturesMode.ExecutionOnly)
+            where TSchema : Schema
+        {
+            schema.Filter = new ExperimentalFeaturesSchemaFilter { Mode = mode };
+            return schema;
+        }
```

Comment:
Something like this:
```suggestion
        public static TSchema EnableExperimentalIntrospectionFeatures<TSchema>(this TSchema schema, ExperimentalIntrospectionFeaturesMode mode = ExperimentalIntrospectionFeaturesMode.ExecutionOnly)
            where TSchema : Schema
        {
            schema.EnableAppliedDirectives = true;
            if (mode == Modes.ExecutionAndIntrospection)
                schema.Filter = new ExperimentalFeaturesSchemaFilter();
            return schema;
        }
```

Reply:
Or maybe:
```cs
schema.ExperimentalFeatures.AppliedDirectives = true;
```

Reply:
Maybe it is worth to move all properties from ISchema into one settings type? Then there will be fewer backward incompatible changes in the future (if we want to keep `ISchema` interface at all).

Reply:
Maybe. What would be on it? Filter and Comparer? Looking through the property list I don’t see much else you’d want to move.

Reply:
I'll put it off for a while. I would like to redesign and remove some properties from the schema.

Reply:
I made changes with a filter. They were spinning in my head and now everything seemed to be in place.